libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -0,0 +1,184 @@
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+"""Test library for query deployment related operations."""
+
+from http import HTTPStatus
+import yaml
+
+from config import PROV_TEST_CFG
+from commons import configmanager
+from commons.constants import LOCAL_SOLUTION_PATH
+from libs.csm.csm_interface import csm_api_factory
+from libs.csm.rest.csm_rest_test_lib import RestTestLib
+
+
+class QueryDeployment(RestTestLib):
+    """This class contains all the Rest API calls for all query deployment operations"""
+
+    def __init__(self):
+        super(QueryDeployment, self).__init__()
+        self.csm_conf = configmanager.get_config_wrapper(
+            fpath="config/csm/test_rest_query_deployment.yaml")
+        self.local_sol_path = LOCAL_SOLUTION_PATH
+        self.resp_dix = None
+        self.resp_sns = None
+        self.prov_deploy_cfg = PROV_TEST_CFG["k8s_prov_cortx_deploy"]
+        self.csm_obj = csm_api_factory("rest")
+
+    def get_solution_yaml(self):
```

Comment:
Can u accept solution.yaml path in parameter. If it is not u can assign below values as default

Reply:
the remote path or local path?

Reply:
remote path

Reply:
will take of this change in next PR.

Reply:
This can now be taken care

Reply:
Pls check the if the modification done is as expected.

Reply:
Instead of declaring with accepting them during __init__ doesn't make sense. So below 2 lines can be deleted and replaced with one of the two solution suggested.

        self.local_sol_path = LOCAL_SOLUTION_PATH
        self.k8s_scripts_path = K8S_SCRIPTS_PATH
--------------------Solution 1 --------------------------------
Can u either accept it as 
def get_solution_yaml(self, local_sol_path =None,k8s_scripts_path =None  ):
 
and if u get None assign them constants.
local_sol_path = LOCAL_SOLUTION_PATH
k8s_scripts_path = K8S_SCRIPTS_PATH

--------------------Solution 2 ---------------------------
Simply use the constants directly as below

remote_sol_path = K8S_SCRIPTS_PATH + 'solution.yaml'

Reply:
done